---
title: "Background"
---


Proteomics is the study of the entire set of proteins that exist in a system (a population, an organism, an organ, a cell, etc.). Identifying and quantifying all these proteins can be used for genetic research, biomarker discovery, population analysis, and countless other applications. In order to study the proteome, we rely heavily on analytical chemistry: specifically separations coupled with mass spectrometry.

The [general workflow](https://en.wikipedia.org/wiki/Bottom-up_proteomics) for a proteomics experiment involves digesting protein into smaller chunks, called peptides, separating these peptides using [liquid chromatography](https://en.wikipedia.org/wiki/High-performance_liquid_chromatography) or [capillary electrophoresis](https://chem.libretexts.org/Bookshelves/Analytical_Chemistry/Supplemental_Modules_(Analytical_Chemistry)/Instrumental_Analysis/Capillary_Electrophoresis), and injecting them into a series of [mass spectrometers](https://en.wikipedia.org/wiki/Mass_spectrometry) which can measure the mass/charge ratio of intact and fragmented peptides. From this information computers can match individual peptides to proteins, and based on signal intensity, the proteins can be quantified.  

![*Proteomics Workflow (Thermo Fisher, planetorbitrap.com)*](Workflows_BottomUpProt-2.jpg)  

As we know, proteins are polymers of individual amino acids, and it is this amino acid sequence (primary structure) that ultimately leads to higher orders of structure and function. Peptides are smaller polymers of amino acids (generally 4-50 residues in length) which (in this context) originate from digested protein.  

![*Example Peptide Structure (Wikipedia)*](Beefy_meaty_peptide.svg.png){width=75%}  

In proteomics, peptides are the analyte of interest, and so we are interested in several of the chemical attributes that they posess: mass, charge, and mass/charge ratio. This last attribute is especially interesting because it is the basis of identification in mass spectrometry, and the basis of separation in capillary electrophoresis. Luckily, both mass and chage of a peptide can be thought of as the summation of the individual masses and charges of the amino acids that make up the peptide. Because there are [20 common amino acids](https://www.sigmaaldrich.com/life-science/metabolomics/learning-center/amino-acid-reference-chart.html), calculating and adding togther their masses or charges becomes a manageable task for a computer to do quickly.  

[There](https://www.peptidesynthetics.co.uk/tools/) [are](https://web.expasy.org/peptide_mass/) [many](http://rna.rega.kuleuven.be/masspec/pepcalc.htm) [online](http://db.systemsbiology.net:8080/proteomicsToolkit/FragIonServlet.html) [tools](http://www.colby.edu/chemistry/NMR/scripts/peptide.html) that can do this type of calculation. While definitely useful, these are not always ideal from an educational standpoint for several reasons. First, they generally work as 'black boxes' where the user puts in a peptide, and numbers are returned. It isn't clear to the user how these values were calculated, and therefore the chemistry behind these calculations isn't fully explored. Second, they can only handle one peptide at a time, or in some cases, one specific protein at a time. In my routine proteomic analysis, I will often have lists of thousands of peptides, and individually using these tools to calculate all of their charges becomes impractical. 

Learning how to use code as a tool for scientific research often seems like a daunting task. There are many publications that include a phrase along the lines of, "The fasta files were digested in silico and probability scores were assigned using an in-house python script". For many students, and even more experienced scientists, this 'in-house' script is a mystery that is conceptually out of reach. However, I believe that there is a false dichotomy of 'knowing programming' or 'not knowing programming' that is prevalent in chemistry, and wet-lab science in general. While understanding very basic coding principles like variables, loops, and functions may be prerequisite information, well-written code can be implemented by almost anyone who is familiar with the underlying scientific principles. In all peer-reviewed journals, these 'in-house' scripts should be published as supporting information, or availible on a platform like git-hub. With only a small amount of experience with code and a good search engine, any researcher should be able to take these availible resourses and apply them to their own questions and data. The goal of this tutorial is to show students that they can take code and use it to answer questions that have real scientific applications, which is an invaluable skill in a research career. 
