<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Step 3</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 45px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h2 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h3 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h4 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h5 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h6 {
  padding-top: 50px;
  margin-top: -50px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">HOME</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="Resources.html">Resources</a>
</li>
<li>
  <a href="Background.html">Background</a>
</li>
<li>
  <a href="Step_1.html">Step 1</a>
</li>
<li>
  <a href="Step_2.html">Step 2</a>
</li>
<li>
  <a href="Step_3.html">Step 3</a>
</li>
<li>
  <a href="Step_4.html">Step 4</a>
</li>
<li>
  <a href="Assignment.html">Assignment</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Step 3</h1>

</div>


<div id="step-3a-calculating-the-charge-for-a-list-of-many-peptides" class="section level2">
<h2>Step 3a: Calculating the charge for a list of many peptides</h2>
<p>Using the <code>pep_charge()</code> function, you were able to calculate the charge for 5 peptides at 3 pH values each. That involved using the function 15 times, which is doable, but slightly tedious. You could have done this same exercise with any of the online tools referenced above, and although it might not have been as educational, you would have achieved the same result. Now say that you have a mixture of 50, or 100, or even 1000 peptides. It suddenly becomes incredibly time-consuming to do all these calculations one at a time. Luckily, because you have the <code>pep_charge()</code> function available in a versatile coding language, you are able to automate some of these calculations.</p>
</div>
<div id="exercise-3a" class="section level2">
<h2>Exercise (3a)</h2>
<p>Here is a list of 50 peptides that you have identified in your sample after a proteomics experiment:</p>
<p><font size="1" color="blue">HSTIFENLANK, HTFMGVVSLGSPSGEVSHPR, HTLNQIDEVK, IDVHLVPDR, IRLENEIQTYR, ITQVLHFTK, IVSSAMEPDREYHFGQAVR, KAEEEHLGILGPQLHADVGDK, KCPLPGTAAFK, KPVDEYKDCHLAQVPSHTVVAR, KPVEEYANCHLAR, KQTALVELVK, KVPQVSTPTLVEVSR, KYLYEIAR, LDELRDEGK, LKYENEVALR, LQHLENELTHDIITK, LSINTHPSQKPLSITVR, LVHVEEPHTETVR, LVNEVTEFAK, LVRPEVDVMCTAFHDNEETFLK, LVRPEVDVMCTAFHDNEETFLKK, LYHSEAFTVNFGDTEEAKK, MRPSTDTITVMVENSHGLR, NECFLQHKDDNPNLPR, PDYSVVLLLR, PSLVEQVFLDK, PVPGHVTVSICR, QEPERNECFLQHK, QEPERNECFLQHKDDNPNLPR, QHIEAIDVR, QNCELFEQLGEYK, QSVEADINGLRR, QEPERNECFLQHKDDNPNLPR, QTALVELVK, RHPDYSVVLLLR, RPCFSALEVDETYVPK, RYLYEIAR, SLHTLFGDK, SPAFTDLHLR, SSNLIILEEHLK, TFHADICTLSEK, THLAPYSDELR, THLPEVFLSK, VDLSFSPSQSLPASHAHLR, VFDEFKPLVEEPQNLIK, VHTECCHGDLLECADDR, VHTECCHGDLLECADDRADLAK, VLANGHAPYSR, VLGYVVEMQPK, VPQVSTPTLVEVSR, VQFELHYQEVK</font></p>
<p>And you want to know the charge of all of them at pH 7.</p>
<p>The code below imports them into R, storing them in a list called <code>peptides</code>. Copy and paste, and then run this in your R console.</p>
<pre class="r"><code>peptides &lt;- list(&quot;HSTIFENLANK&quot;, &quot;HTFMGVVSLGSPSGEVSHPR&quot;, &quot;HTLNQIDEVK&quot;, &quot;IDVHLVPDR&quot;, &quot;IRLENEIQTYR&quot;, &quot;ITQVLHFTK&quot;, &quot;IVSSAMEPDREYHFGQAVR&quot;, &quot;KAEEEHLGILGPQLHADVGDK&quot;, &quot;KCPLPGTAAFK&quot;, &quot;KPVDEYKDCHLAQVPSHTVVAR&quot;, &quot;KPVEEYANCHLAR&quot;, &quot;KQTALVELVK&quot;, &quot;KVPQVSTPTLVEVSR&quot;, &quot;KYLYEIAR&quot;, &quot;LDELRDEGK&quot;, &quot;LKYENEVALR&quot;, &quot;LQHLENELTHDIITK&quot;, &quot;LSINTHPSQKPLSITVR&quot;, &quot;LVHVEEPHTETVR&quot;, &quot;LVNEVTEFAK&quot;, &quot;LVRPEVDVMCTAFHDNEETFLK&quot;, &quot;LVRPEVDVMCTAFHDNEETFLKK&quot;, &quot;LYHSEAFTVNFGDTEEAKK&quot;, &quot;MRPSTDTITVMVENSHGLR&quot;, &quot;NECFLQHKDDNPNLPR&quot;, &quot;PDYSVVLLLR&quot;, &quot;PSLVEQVFLDK&quot;, &quot;PVPGHVTVSICR&quot;, &quot;QEPERNECFLQHK&quot;, &quot;QEPERNECFLQHKDDNPNLPR&quot;, &quot;QHIEAIDVR&quot;, &quot;QNCELFEQLGEYK&quot;, &quot;QSVEADINGLRR&quot;, &quot;QEPERNECFLQHKDDNPNLPR&quot;, &quot;QTALVELVK&quot;, &quot;RHPDYSVVLLLR&quot;, &quot;RPCFSALEVDETYVPK&quot;, &quot;RYLYEIAR&quot;, &quot;SLHTLFGDK&quot;, &quot;SPAFTDLHLR&quot;, &quot;SSNLIILEEHLK&quot;, &quot;TFHADICTLSEK&quot;, &quot;THLAPYSDELR&quot;, &quot;THLPEVFLSK&quot;, &quot;VDLSFSPSQSLPASHAHLR&quot;, &quot;VFDEFKPLVEEPQNLIK&quot;, &quot;VHTECCHGDLLECADDR&quot;, &quot;VHTECCHGDLLECADDRADLAK&quot;, &quot;VLANGHAPYSR&quot;, &quot;VLGYVVEMQPK&quot;, &quot;VPQVSTPTLVEVSR&quot;, &quot;VQFELHYQEVK&quot;)</code></pre>
<p>You may not know exactly how to make your function run individually on many different peptides, but luckily you found another published script that does something related. In this case, the function <code>pep_length()</code> calculates the number of amino acids in a sequence. Then, below that, the author wrote a <code>for</code> loop that goes through a list of peptides (just like the one you have!) and calculates and prints the length for each one. Copy and paste the code below to your ‘.R’ file, and run it.</p>
<pre class="r"><code>#function that returns the number of amino acids in a peptide
pep_length &lt;- function(peptide) {
  length &lt;- nchar(peptide)
  return(length)
}


#loop through a list of peptides and print out the peptide followed by the number of amino acids
for (pep in peptides) {
  #print out the peptides, one at a time. &#39;pep&#39; is the variable that is a different peptide each loop
  print(pep)
  
  #calculate the length of the peptide
  peptide_length &lt;- pep_length(peptide = pep)
  
  #print out the peptide length
  print(peptide_length)
}</code></pre>
<p>If you ran the peptides list command earlier and this new block of code, you should see an output that printed each peptide followed by the number of amino acids it contains. Here are what the first three printouts should look like:</p>
<pre><code>## [1] &quot;HSTIFENLANK&quot;
## [1] 11
## [1] &quot;HTFMGVVSLGSPSGEVSHPR&quot;
## [1] 20
## [1] &quot;HTLNQIDEVK&quot;
## [1] 10</code></pre>
<p>Now you want to do essentially the same thing, but instead of printing out the length, print out the charge at pH 7. How can you modify the code to print out the charge? (Some hints are given below in white font… highlight the blank space below to read them.) <font color="purple">Take a screenshot of your modified code, and put it in the assignment.</font></p>
<p><font color="white">Hint 1: The modification should take place within the <code>for</code> loop.<br />
Hint 2: The modification involves exchanging one function for a different one.<br />
Hint 3: <code>pep_length()</code> is a function, just like <code>pep_charge()</code>. In this case, both include <code>peptide = pep</code>, but <code>pep_charge()</code> also takes an additional argument.</font></p>
<p>If it worked correctly, you might get an output that includes the correct charges, but also includes the dataframe used to calculate the charge for each peptide. That is a lot of information, and not necessary in this case. Go back to your original <code>pep_charge()</code> function from exercise 1. Find the line that prints out the <code>net_charge_df</code> and delete it or comment it out (putting a <code>#</code> symbol in front of a line of code tells the computer to ignore it - this is referred to as ‘commenting out a line’). Re-run that original <code>pep_charge()</code> function to save the change, and then re-run your ‘for loop’ to print out the values. Is that more manageable?</p>
</div>
<div id="step-3b-predicting-mz-values-for-peptides-in-maldi-experiments" class="section level2">
<h2>Step 3b: Predicting m/z values for peptides in MALDI experiments</h2>
<p>Matrix Assisted Laser Desorption Ionization Time of Flight (MALDI-TOF) mass spectrometry is an excellent way to quickly obtain mass values for analytes in a solution. In this technique, the sample is combined with a ‘matrix’ (usually some sort of organic acid). This matrix-sample combination is dried on a plate, and placed in the instrument. A laser excites the sample and desorbs both matrix and sample from the plate. In this high energy cloud, some analyte obtains an extra proton from the matrix and becomes positively charged. The m/z value of this analyte is measured by the time it takes to reach an ion detector, after being accelerated by an applied voltage. Analytes with smaller m/z values will travel faster than those with larger m/z values.</p>
<div class="figure">
<img src="MALDI_TOF_EN.png" style="width:50.0%" alt="" />
<p class="caption"><em>MALDI-TOF Schematic (Wikipedia)</em></p>
</div>
</div>
<div id="exercise-3b" class="section level2">
<h2>Exercise (3b)</h2>
<p>You run each of the fractions from exercise 2 on a MALDI-TOF instrument, and obtain mass spectra, which show signal at specific m/z values. You want to calculate the expected m/z values for each of your peptides to see if they are present. Luckily, you find 2 more helpful published R functions. The first takes a peptide, and returns the mass of the peptide in Daltons. The second takes a mass and a charge value and returns m/z. Copy and paste both of these functions into R and run them.</p>
<pre class="r"><code>#calculate peptide mass
#create_dictionary. Each one letter code corresponds to a mass value. Must be run before the function
one_letter &lt;- c(&#39;A&#39;,&#39;R&#39;,&#39;N&#39;,&#39;D&#39;,&#39;C&#39;,&#39;E&#39;,&#39;Q&#39;,&#39;G&#39;,&#39;H&#39;,&#39;I&#39;,&#39;L&#39;,&#39;K&#39;,&#39;M&#39;,&#39;F&#39;,&#39;P&#39;,&#39;S&#39;,&#39;T&#39;,&#39;W&#39;,&#39;Y&#39;,&#39;V&#39;)
masses &lt;- list(71.03711381, 156.1011111, 114.0429275, 115.0269431, 103.0091845, 129.0425931, 128.0585775,
               57.02146374, 137.0589119, 113.084064, 113.084064, 128.0949631, 131.0404846, 147.0684139,
               97.05276388, 87.03202844, 101.0476785, 186.079313, 163.0633286, 99.06841395)
names(masses) &lt;- one_letter

#the function to calculate the mass
pep_mass &lt;- function(peptide){
  
  #convert the string to uppercase
  peptide &lt;- toupper(peptide)
  
  #split the peptide into individual letters. [[1]] is required to select the individual values as opposed to a list
  splitpep &lt;- strsplit(peptide,&#39;&#39;)[[1]]
  
  #create a variable to store the mass. Set it to 0 to start
  total_mass &lt;- 0.0
  
  #loop through each residue (&#39;AA&#39; in this case) in the peptide
  #first ensure that it is within the list of one letter codes above (by checking that it isn&#39;t null)
  #then add the corresponding mass for that one letter code to the total mass. masses[[AA]] accesses the mass corresponding to the specific amino acid
  for (AA in splitpep) {
    if (!is.null(masses[[AA]])) {
      total_mass &lt;- total_mass + masses[[AA]]
    } 
  }
  
  #add the mass of H2O (for the n and c terminus... the masses used assume peptide bonds between amino acids)
  total_mass &lt;- total_mass + 18.010565
    
  #return the total mass
  return(total_mass)
}

#A function that calculates m/z given a peptide and a pH
m_over_z &lt;- function(peptide, pH = 7) {
  peptide_mass &lt;- pep_mass(peptide)
  peptide_charge &lt;- pep_charge(peptide, pH = 7)
  mass_charge_ratio &lt;- peptide_mass / peptide_charge
  return(mass_charge_ratio)
}</code></pre>
<p>Lets try it on one of our favorite peptides, DIAK. You should run this in your own console as well.</p>
<pre class="r"><code>m_over_z(&quot;DIAK&quot;, pH = 7)</code></pre>
<pre><code>## [1] -18971.03</code></pre>
<p>Hmmm that output seems very high (and negative). What could be wrong? Lets try calculating charge and mass separately.</p>
<pre class="r"><code>pep_charge(&quot;DIAK&quot;, pH = 7)</code></pre>
<pre><code>## [1] -0.02347019</code></pre>
<pre class="r"><code>pep_mass(&quot;DIAK&quot;)</code></pre>
<pre><code>## [1] 445.2536</code></pre>
<p>The mass looks reasonable, but the charge is very close to zero (we actually knew this already from exercise 1). It turns out, when we are dividing mass / charge, we are dividing by a number that is essentially zero, causing the m/z value to skyrocket.</p>
<p>If you think about MALDI however, we don’t actually know what the pH is, and so assuming it to be 7 is probably not accurate. In addition, based on your extensive knowledge of MALDI with peptides, you remember that almost all peptides end up with a +1 charge (due to one proton) when they are ionized with this technique. How could you modify the <code>m_over_z()</code> function to make it applicable to MALDI? The important aspects are:</p>
<ul>
<li>The charge will always be +1</li>
<li>The mass will be the mass of the peptide <em>plus the mass of a proton</em> which is approximately 1.008 Da.</li>
</ul>
<p><font color="purple">Take a screenshot of your modified <code>m_over_z()</code> function and paste it in your assignment. What are the expected MALDI m/z values for each peptide from Exercise 2?</font></p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
